README.md

```diff
@@ -1,2 +1,150 @@
+
+## Abstract
+
+This specification describes "mailto" [DID Method] that
+conforms to the core [DID-CORE] specification. The method can be used independent of any central source of truth, and is intended for bootstrapping secure interaction between two parties that can span across arbitrary number of devices. It is suitable for long sessions that need to operating under network partitions.
+
+## Status of This Document
+
+This document is draft of a potential specification. It has no official standing of any kind and does not represent the support or consensus of any standards organization.
+
+## Introduction
+
+### Overview
+
+> This section is non-normative.
+
+Most documentation about decentralized identifiers (DIDs) describes them as identifiers that are rooted in a public source of truth like a blockchain, a database, or similar. This publicness lets arbitrary parties resolve the DIDs to an endpoint and keys. It is an important feature for many use cases.
+
+However, the vast majority of interactions between people, organizations, and things have simpler requirements. When Alice(Corp|Device) and Bob want to interact, there are exactly and only 2 parties in the world who should care: Alice and Bob. Instead of arbitrary parties needing to resolve their DIDs, only Alice and Bob do.
+
+One to one interactions are excellent fit for [did:key] identifiers, however they suffer from key discovery problem and introduce additional challenges when interaction sessions span across more than two devices.
+
+Mailto DIDs are designed to be used in conjunction with [did:key] and facilitate bootstrapping sessions between two parties that span across multiple devices.
+
+Mailto DIDs are more accessible alternative to [did:web] and [did:dns] because a lot more people have an email address than there are people with [did:web] or [did:dns] identifier or skills to acquire one.
+
+Mailto [did document]s are also verifiable offline, when [domain key][] of the email address is known, implying significantly less network requests than with most other DIDs.
+
+## The `did:mailto` Format
+
+The format for the `did:key` method conforms to the [DID-CORE] specification and is encoding of the [email address]. It consists of the `did:mailto:` prefix, followed by the `domain` part of an email address, `:` character and percent encoded `local-part` of the email address.
+
+The ABNF definition can be found below. The formal rules describing valid `domain-name` syntax are described in [RFC1035], [RFC1123], [RFC2181]. The `domain-name` and `user-name` corresponds to `domain` and `local-part` respectively of the email address described in [RFC2822]. All "mailto" DIDs MUST conform to the DID Syntax ABNF Rules.
+
+```abnf
+did       = "did:mailto:" domain-name ":" user-name
```

Comment:
My rationale was:

1. Unfortunately `new URL(“did:mailto:alice@web.mail”)` does not get escaped, so it’s seems more likely to encounter malformed versions unless encoding is distinct enough
2. DIDs seem to love `:` delimiter, `did:web` even goes as far as replacing `/` with them, so this seemed reasonable
3. And yeah I find it harder to read

Above said I’m not feeling strongly about this

Reply:
fwiw I also kinda prefer the simplicity of just `did:mailto:pct-encoded-email-address`

We could also establish some norms around, regardless of did mailto, what to do if someone sends you a did that is not actually a valid did, e.g. `did:mailto:bengo@dag.house` or just `did:whatever:bengo@dag.house`. We could just say "percent-encode any colon-delimited segment in `method-specific-id` that isn't already in the valid alphabet, and implement that in `@ipld/dag-ucan` or something in `DID.from`

```
DID.from('did:whatever:bengo@dag.house')
// did:whatever:bengo%40dag.house
```

This is a little inspired by how web browsers in practice don't always show you the perfectly encoded URL in the URL bar. It does its best to decode the url-like strings you put in there. If we do something like that, maybe we can avoid the annoyance of `%40` instead of `@` because a lot of our tooling will accept the `@` and only throw when it really needs to be strict about being a compliant did.

Reply:
Just wanted to add as an additional perspective: we use DIDs as domain-style origins within a human-author-able link syntax. For example, a user _may_ write:

`@did:key:.../foo` 

...or...

`@did:web:example.com/foo`

In the case of `did:key`, it is not convenient for a human to write by hand w/o copy-pasting the DID. In practice, we implement a petname system and alias DIDs by petnames that are easy for humans to write.

But, with `did:web` (and perhaps `did:mailto`), it is more plausible that a human will author a DID by hand, or read a link with a bare (not-petname-aliased) DID in it. And, for that case it would be preferable for `did:mailto` to be written and rendered without a %-encoded `@` in the email address.

Reply:
> it would be preferable for did:mailto to be written and rendered without a %-encoded @ in the email address.

I think we can accomodate this by in practice users just typing `did:mailto:bengo@dag.house` but anywhere that gets that string as input and is casting it to a valid did needs to pct-encode the `:`-delimited segments to match syntax requirements of DIDs.

i.e. if the method-specific-identifier contains a `@`, you can tell it's not a valid DID, but that it wants to be a `did:mailto`, so software can deterministically get to a valid did by pct-encoding the email address.